porch/pkg/git/git.go

```diff
@@ -486,7 +509,12 @@ func (r *gitRepository) loadPackageRevision(ctx context.Context, version, path s
 
 	var ref *plumbing.Reference = nil // Cannot determine ref; this package will be considered final (immutable).
 
-	packageRevision, err := krmPackage.buildGitPackageRevision(ctx, version, ref)
+	description, err := getPkgDescription(commit, path)
```

Comment:
I would expect that the description was a parameter to the `loadPackageRevision` function, as it is one of the properties that uniquely identifier a packagerevision (although in the draft state). Or how do we handle it since we effectively end up with somewhat different properties depending on whether a package is published or not.

Reply:
Thanks for pointing this out, I took some more time to go through this code today.

It looks like the `version` that gets passed through here (through `GetPackageRevision` and then to `loadPackageRevision`) isn't actually necessarily a revision number, it can be any git ref (a branch name, tag, commit SHA, etc). It is `loadPackageRevision`'s job to deference this into a commit and package revision number. 

I think `loadPackageRevision` is incorrectly assuming that the version passed in is always just the revision number, which leads to incorrect behavior. For example, I invoked this code using something like `kpt alpha rpkg clone https://github.com/natasha41575/blueprints.git/foo -ndefault --repository=deployments test --ref=drafts/foo/v4 --revision=v14` to see what would happen when I passed in a ref with a draft prefix. Stepping through, I can see that the current `loadPackageRevision` actually returns a git package revision with a revision field of `drafts/foo/v4` in this case. Similarly, I ran `kpt alpha rpkg clone https://github.com/natasha41575/blueprints.git/foo -ndefault --repository=deployments test2 --ref=foo/v3 --revision=v15` (where `foo/v3` corresponds to a published tag). Stepping through `loadPackageRevision` again, and I can see that this results in a git package revision with a revision field of `foo/v3`. That doesn't seem correct to me.  

This code is only invoked when we clone directly from git and we are trying to fetch the package revision from upstream, so I don't think the incorrect revision/description makes a difference here. But I updated `loadPackageRevision` in my most recent commit so that we do actually parse the passed in ref to the right revision/description based on the prefix of the ref.